Updated drop-in files section (bsc#1212777)

tbazant opened the pull request:
### PR creator: Are there any relevant issues/feature requests?

* bsc#1212777
* jsc#...


### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 next/openSUSE Leap next *(current `main`, no backport necessary)*
  - [x] SLE 15 SP4/openSUSE Leap 15.5
  - [x] SLE 15 SP4/openSUSE Leap 15.4
  - [x] SLE 15 SP3/openSUSE Leap 15.3
  - [x] SLE 15 SP2/openSUSE Leap 15.2
  - [x] SLE 15 SP1
- SLE 12
  - [x] SLE 12 SP5
  - [x] SLE 12 SP4

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

Werkov commented on xml/systemd.xml:
```diff
@@ -1485,51 +1485,75 @@ UMask=007</screen>
+        or <emphasis>drop-ins</emphasis>. Drop-ins let you extend the
+        configuration of unit files without having to edit or override the unit
+        files themselves. Drop-ins have higher precedence over main
+        configuration files. They are stored in the following directories:
```
There are two classes of config files: _unit_ files and daemon config files.
So drop-ins let you extend both classes of configs.

Werkov reviewed (changes requested):
Looks good (@fbuihuu may refine the numeric ranges but I think these are the latest values).
I'd only emphasize more the difference between unit files (with their drop-ins) and daemon config files (with their drop-ins).

fbuihuu commented:
> I'd only emphasize more the difference between unit files (with their drop-ins) and daemon config files (with their drop-ins).

I agree.

For viewing/editing unit files there're `systemctl cat`/`systemctl edit` respectively but for config files there's only `systemd-analyze cat-config` currently.

fbuihuu commented:
> Looks good (@fbuihuu may refine the numeric ranges but I think these are the latest values).

```
[0 - 19]  : systemd upstream
[20 - 25] : systemd SUSE
[26 - 29] : systemd packages
[30 - 49] : unused
50 : `systemctl set-property` for unit files only
```
I'm not sure where these ranges are coming from. They look OK although I'm not sure to understand why we have such small ranges reserved for downstream and packages. Also why `[30 - 49]` is wasted.

Maybe we shouldn't go into such level of details if the doc is aimed at users/customers, simply stating that they should use the range `[60 - 99]` only might be enough, no ?

Werkov commented on xml/systemd.xml:
```diff
@@ -1485,51 +1483,78 @@ UMask=007</screen>
       </para>
+      <variablelist>
+        <varlistentry>
+          <term><filename>/etc/systemd/*.conf.d/</filename></term>
```
Paths for unit files:
/etc/systemd/system/*.service.d/

Werkov commented on xml/systemd.xml:
```diff
@@ -1485,51 +1483,78 @@ UMask=007</screen>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term><filename>/usr/lib/systemd/*.conf.d/</filename></term>
```
Path for unit files:
/usr/lib/systemd/system/*.service.d/

See `systemd.unit(5)` for full _unit search path_.

Werkov commented on xml/systemd.xml:
```diff
@@ -1485,51 +1483,78 @@ UMask=007</screen>
-        For minor changes of a configuration file, use so-called drop-in files.
-        Drop-in files let you extend the configuration of unit files without
-        having to edit or override the unit files themselves.
+        For minor changes of a configuration file, use so-called drop-in files
```
Maybe drop _minor_?
As a drop-ins may be the preferred way of config modification in the future.
Does it make sense, @fbuihuu ?

tbazant replied:
```suggestion
        For changes of configuration files, use so-called drop-in files
```

Werkov reviewed (changes requested):
I've added a few points for unit/non-unit drop-ins distinction.

The ranges aren't changed in this PR, that is what is in SLE documentation since some time (and they're consistent with [openSUSE doc](https://en.opensuse.org/openSUSE:Systemd_packaging_guidelines#Shipping_unit_file_drop-ins)).

I notice the added commit includes also some stylistic changes -- I didn't look at them, please add them to a separate commit (next time ;-) for easier review.

dariavladykina reviewed (commented):
Hi, please note some suggestions here. Thanks!

tbazant reviewed (commented):
applied suggestions

tbazant commented:
> I've added a few points for unit/non-unit drop-ins distinction.
> 
> The ranges aren't changed in this PR, that is what is in SLE documentation since some time (and they're consistent with [openSUSE doc](https://en.opensuse.org/openSUSE:Systemd_packaging_guidelines#Shipping_unit_file_drop-ins)).
> 
> I notice the added commit includes also some stylistic changes -- I didn't look at them, please add them to a separate commit (next time ;-) for easier review.

my bad, i thought technical matters are complete, then i asked for style review which is out workflow. any additions to the current state BTW? or should i merge finally?

fbuihuu commented:
> The ranges aren't changed in this PR, that is what is in SLE documentation since some time (and they're consistent with [openSUSE doc](https://en.opensuse.org/openSUSE:Systemd_packaging_guidelines#Shipping_unit_file_drop-ins)).

Maybe it's still possible the change the ranges affected to downstream. I currently don't see why we affected such small ranges fo SUSE purposes. Something like the following seems better:

```
[0 - 19] : upstream systemd
[20 - 29]: SUSE systemd
[30 - 39]: SUSE packages
[40 - 49]: 3rd party packages
```

BTW the openSUSE doc states:

> Hence by using a two-digit number above range [0-29], users will be assured that none of the drop-in files shipped by SUSE will override users own drop-in files in the future.

I think this should be corrected to claim that only drop-ins with `priority >= 50`  are safe from a user POV.

fbuihuu commented:
> 
> ```
> [0 - 19] : upstream systemd
> [20 - 29]: SUSE systemd
> [30 - 39]: SUSE packages
> [40 - 49]: 3rd party packages
> ```

I updated the openSUSE doc to use these ranges. SUSE systemd/packages ranges having a short range doesn't make really sense.

I think the SLE doc should do the same. I don't think these "system" ranges are used by customers.

tbazant commented:
> > ```
> > [0 - 19] : upstream systemd
> > [20 - 29]: SUSE systemd
> > [30 - 39]: SUSE packages
> > [40 - 49]: 3rd party packages
> > ```
> 
> I updated the openSUSE doc to use these ranges. SUSE systemd/packages ranges having a short range doesn't make really sense.
> 
> I think the SLE doc should do the same. I don't think these "system" ranges are used by customers.

i've updated the SLE sources accordingly, will be published soon
